src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -703,6 +804,57 @@ TrackPointer RekordboxPlaylistModel::getTrack(const QModelIndex& index) const {
     qDebug() << "RekordboxTrackModel::getTrack";
 
     TrackPointer track = BaseExternalPlaylistModel::getTrack(index);
+    QString location = index.sibling(index.row(), fieldIndex("location")).data().toString();
+
+    // The following code accounts for timing offsets required to
+    // correctly align timing information (cue points, loops, beatgrids)
+    // exported from Rekordbox. This is caused by different MP3
+    // decoders treating MP3s encoded in a variety of different cases
+    // differently. The mp3guessenc library is used to determine which
+    // case the MP3 is clasified in. See the following PR for more
+    // detailed information:
+    // https://github.com/mixxxdj/mixxx/pull/2119
+
+    int timingOffset = 0;
+
+    if (location.toLower().endsWith(".mp3")) {
+        int timingShiftCase = mp3guessenc_timing_shift_case(location.toStdString().c_str());
+
+        qDebug() << "Timing shift case:" << timingShiftCase << "for MP3 file:" << location;
+
+        switch (timingShiftCase) {
+#ifdef __COREAUDIO__
```

Comment:
I think what we want here is the offset for the decoder the file will be read with, which IIUC is not quite the same as what options Mixxx is built with. @uklotzde could you take a look at this?

Reply:
From what I can see, tracks are completely decoupled from the audio sources that loaded their audio data, so if I understand correctly there is currently no way of determining what decoder has been used to read a given track without serious alterations to the way in which tracks are loaded. And track data can be accessed before tracks are loaded to a deck when their audio data is provided by an audio source. In this scenario, if tracks were altered to include audio source information, it wouldn't exist yet. Happy to explore other suggestions of determining the MP3 decoder of a given track at any time.

Reply:
Is using SoundSourceProxy a reasonable alternative? eg.
```
SoundSourceProxy::findSoundSourceProviderRegistrations(track->getFileInfo().toUrl());
```
Then iterating though each provider and determining which decoder will be used by comparing returned provider priority hints.

Reply:
The list is already ordered according to priorities. Usually, the first provider will succeed to open the file, but this is not guaranteed. You won't know until you actually try it. Just pick the first.